test/libndt7_test.cpp

```diff
@@ -83,13 +83,18 @@ TEST_CASE("Client::parse_ws_url() table tests") {
       .url = "://",
       .want = {.scheme = "", .host = "", .port = "", .path = ""},
     },
-    /*
-    TODO(soltesz): support parsing IPv6 hosts.
     {
-      .url = "ws://[::1]/test?foo",
-      .want = {.scheme = "ws", .host = "[::1]", .port = "80", .path = "/test?foo"},
+      .url = "ws://127.0.0.1:8888/test?foo",
+      .want = {.scheme = "ws", .host = "127.0.0.1", .port = "8888", .path = "/test?foo"},
+    },
+    {
+      .url = "wss://[abc::1]/test?foo&x=1",
+      .want = {.scheme = "wss", .host = "[abc::1]", .port = "443", .path = "/test?foo&x=1"},
+    },
+    {
+      .url = "ws://[abc::1]:9999/test?foo&x=1",
+      .want = {.scheme = "ws", .host = "[abc::1]", .port = "9999", .path = "/test?foo&x=1"},
     },
```

Comment:
Please, add a test case where the IPv6 address is like `a:b:c:d` (i.e., a case where there are no `::` characters inside it. I have manually verified your regexp passes this test case, but I'd rather a test case anyway.

Reply:
See https://regex101.com/r/31uSa5/1 for the regexp I tested.